### ISSUE TITLE
React/MUI: Focusing children and then the parent again breaks behavior

### ISSUE THREAD

**Issue by tngkavsek** [nvaccess/nvda#13622]
### Steps to reproduce:
I created a sandbox with a MUI-Accordion to demonstrate my issue, in short:

1. Activate NVDA
2. create a mui-accordion in react containing at least one <AccordionAction>
3. focus element using TAB-Key (Pressing enter now opens and closes the accordion as desired)
4. focus the first action by using TAB again (now, pressing enter activates the tab and opens the tile)
5. with shift+tab, focus the Summary again, then press enter (now, again the button is pressed and the accordion is opened)

https://codesandbox.io/s/accordion-demo-p0pun2?file=/src/App.tsx

### Actual behavior:
1. When focusing a child element and then the parent again, it should only open the accordion and not focusing the child again
2. When switching between the AccordionSummary and One Of the AccordionActions, the matching elements should be read by NVDA.

### Expected behavior:
This happens only when NVDA is active:
1. After focusing the child and then the parent again, it focusses the child when pressing enter. 
2. The switch between parent and child is not recognized by NVDA

#### NVDA installed/portable/running from source:
NVDA installed

#### NVDA version:
2021.3.5 (latest)

#### Windows version:
Windows 11 Pro Build 22000.613

#### Name and version of other software in use when reproducing the issue:
Browser: Brave (Chrome)
Frameworks: React + https://mui.com/

#### Other information about your system:
Tested with other screen readers and it worked as desired

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Updated after the issue occurred, same error in last version. 

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by alexarnaud** [comment]
Hello,

Thanks a lot for reporting this issue.

I just checked the HTML result as I'm not a React guru.

It seems there is a button inside a button. The parent element of the button The div containing the action button has also role button and a label, which is as of my understanding, not the way the accordion is recommended to be implemented, [see the W3C example here](https://www.w3.org/TR/wai-aria-practices-1.1/#accordion).

Looking at Stackoverflow linking the W3C, it is not valid HTML:
[Can I nest button inside another button?](https://stackoverflow.com/questions/39386497/can-i-nest-button-inside-another-button)

Hope that help.

**Comment by tngkavsek** [comment]
I think it helps. I will close this issue for now because I don't think there is anything to do for you. Thanks for the fast reply!
